### PR TITLE
change PayForMessage to PayForData

### LINKED ISSUE #269: Rename `PayForMessage` to `PayForData`
For a hopefully clearer naming scheme, we can change Message -> data in both `MsgPayForMessage` and `WireMsgPayForMessage`

### PR DESCRIPTION
**Pull request by quinle**
Close #269

### CODE REVIEW

**Comment by quinle** [comment]
@evan-forbes do you know why the Lint fail?

**Comment by evan-forbes** [comment]
Sorry for the delay on this, I think the linter is failing because it auto updates to the latest version of go. We've fixed it in core by explicitly picking a version. See https://github.com/celestiaorg/celestia-core/pull/637/commits/2a5a3044b97410baafff07b1ef2f180b505fdc7f

I can try this fix today and see if we can fix, or feel free to as well @quinle 

Thanks again for taking this on @quinle, appreciate it 🙂

**Comment by evan-forbes** [comment]
started to fix the linter in #273, so we can merge that PR first, and then merge this one

**Review by liamsi — CHANGES_REQUESTED** [review]
h/t to @Bidon15:
the file(s) should also be renamed btw: 
- x/payment/types/payformessage.go
- x/payment/types/payfordata.go

File: x/payment/types/tx.pb.go [hunk 1]
```diff
@@ -29,28 +29,28 @@ var _ = math.Inf
 // proto package needs to be updated.
 const _ = proto.GoGoProtoPackageIsVersion3 // please upgrade the proto package
 
-// MsgWirePayForMessage describes the format of data that is sent over the wire
-// for each PayForMessage
-type MsgWirePayForMessage struct {
```
Comment (liamsi) [inline x/payment/types/tx.pb.go:34]: Are these re-generated or manually renamed as well?

**Review by evan-forbes — CHANGES_REQUESTED** [review]
Thanks again @quinle for doing this, there are only a few remaining things that we need to do before we can merge.

see the `URLMsgWirePayforMessage` and `URLMsgPayforMessage` consts in x/payments/types, we should rename those var names to reflect the new naming scheme

We also have to recompile the proto files after we change anything in them, this results in a change to some one of the generated files. Sometimes getting the protofiles to compile can be a chore, as it requires installing docker and then calling `make proto-gen`. Let me know if you're having trouble :slightly_smiling_face:

**Comment by quinle** [comment]
@evan-forbes I have changed the constants URLPayforMessage to URLPayforData but I'm not sure how to recompile the protofiles. Can you direct me to some tutorial on this?

**Comment by evan-forbes** [comment]
> I have changed the constants URLPayforMessage to URLPayforData but I'm not sure how to recompile the protofiles. Can you direct me to some tutorial on this?

if you have docker installed, you should just be able to call
`make proto-gen` in the root of the celestia-app repo

**Comment by YazzyYaz** [comment]
Hi @quinle thank you for this PR. Is there anything we can do to help bring it past the finish line? Really appreciate all the good work you're doing.

**Comment by liamsi** [comment]
@quinle do you still plan to tackle this? Otherwise, I'm happy to push the requested changes here.

**Comment by quinle** [comment]
> @quinle do you still plan to tackle this? Otherwise, I'm happy to push the requested changes here.

@liamsi I'm on it this weekends. Last week was rough for me.
@evan-forbes I thought I have to build with the Dockerfile in celestia-app but I don't have to.
However, when I run `make proto-gen`, I got this error:
```
unknown command "protoc" for "buf"
make: *** [proto-gen] Error 1
```
What could be the issue for this?

**Comment by quinle** [comment]
> @quinle do you still plan to tackle this? Otherwise, I'm happy to push the requested changes here.

Sorry last week was rough for me. I will try to wrap this up this weekends. 
@evan-forbes , at first I thought I have to build from Dockerfile in celestia-app but turn out it is not.
However, when I run `make proto-gen`, I got the following error:
```
unknown command "protoc" for "buf"
make: *** [proto-gen] Error 1
```

I know that I'm missing a module/library, I tried https://github.com/bufbuild/buf but it seems not to be the case. 
What could be the problem?

**Comment by evan-forbes** [comment]
that's a weird error, as you shouldn't have to have that installed, since it is already installed in the docker container that is being used to recompile the protofiles. 

I can try removing them from my machine and recompiling. If for some reason you do have to have them installed, we don't have to add them to the go.mod as described, just installed on your local machine.

if it keeps giving us trouble, I can compile for us and add a commit. Thanks for staying up to date with master. I can also look into the failing CI, as it should all be passing after updating

**Comment by liamsi** [comment]
I ran into the same issue as @quinle. This could be related to this change: https://github.com/tendermint/tendermint/pull/7975

**Comment by evan-forbes** [comment]
update: I can still generate the proto files fine, but I think that's because I have an old cached docker image on my machine. If we want to update, then we will have to use updated scripts from the cosmos-sdk. Ran into a blocker there, see https://github.com/cosmos/cosmos-sdk/issues/11668

will update here after we fix https://github.com/celestiaorg/celestia-app/issues/295

**Comment by quinle** [comment]
could you help to push the compiled change? Let's finish this pull and I'm excited to dig into other issues 💯

**Review by evan-forbes — APPROVED** [review]
went ahead and fixed the remaining changes while we fix #295 

thanks for doing this @quinle, glad we get to add you to our contributor list!

**Review by liamsi — COMMENTED** [review]
Should we also rename the files as well: payformessage.go, wirepayformessage.go etc?

**Comment by evan-forbes** [comment]
> Should we also rename the files as well: payformessage.go, wirepayformessage.go etc?

good catch! https://github.com/celestiaorg/celestia-app/pull/271/commits/9a78dabc3bfabdea0dbd9f25f9537fabea62793c

**Review by Bidon15 — CHANGES_REQUESTED** [review]
Needs renaming for file: 
- `x/payment/client/cli/wirepayformessage.go`

Otherwise, lgtm 💪

**Review by evan-forbes — APPROVED** [review]
alighty, I think this is finally good to go

**Review by liamsi — APPROVED** [review]
👍🏼

**Comment by YazzyYaz** [comment]
@quinle thank you so much for your contribution. Let me know if you'd like to discuss more involvement and contributions to Celestia's github tickets.

**Comment by quinle** [comment]
> @quinle thank you so much for your contribution. Let me know if you'd like to discuss more involvement and contributions to Celestia's github tickets.

Yes, I would love to involve! Do you have recommendations on the ticket I should work next?

**Pull request merged**